### ISSUE TITLE
asTransaction() causes Amount hashing failure in v0.18.1+ 

### ISSUE THREAD

**Issue by Kirol54** [aergoio/herajs#95]
up until v0.17.1 the following line worked : 
```
 const callTx = contract[funcName](...args).asTransaction({
    nonce: nonce + 1,
    from: identity.address,
    chainIdHash: chainIdHash
  }); 
callTx.sign = await signTransaction(callTx, identity.keyPair);
```
from v0.18.1 we get the following error 

```
index.js:1 Error: Can only hash amounts provided in the base unit (aer), not '0 aergo'. Convert to aer or remove unit.
    at herajs-crypto.umd.js:20398
    at Generator.next (<anonymous>)
    at asyncGeneratorStep (herajs-crypto.umd.js:9)
    at _next (herajs-crypto.umd.js:31)
    at herajs-crypto.umd.js:38
    at new Promise (<anonymous>)
    at herajs-crypto.umd.js:27
    at _hashTransaction (herajs-crypto.umd.js:20428)
    at hashTransaction (herajs-crypto.umd.js:20380)
    at herajs-crypto.umd.js:20475
    at Generator.next (<anonymous>)
    at asyncGeneratorStep (herajs-crypto.umd.js:9)
    at _next (herajs-crypto.umd.js:31)
    at herajs-crypto.umd.js:38
    at new Promise (<anonymous>)
    at herajs-crypto.umd.js:27
    at _signTransaction (herajs-crypto.umd.js:20478)
    at signTransaction (herajs-crypto.umd.js:20465)
    at callSmartContract (herajs.js:81)
    at async registerUser (register.js:24)
```

**Comment by aspiers** [comment]
This prevents us from testing the fix to #94.

**Comment by aspiers** [comment]
We found that changing

https://github.com/aergoio/herajs/blob/b4ddc8533cbaf76906c803e1034de22a209a0e16/packages/client/src/models/contract.ts#L63

to

```js
        amount: new Amount("0 aer"),
```

fixed it, but this is kind of a guess since we don't fully understand the reason for the different units :-)

**Comment by aspiers** [comment]
BTW this failure suggests that the test suite is missing coverage for this usage?

**Comment by graup** [comment]
Hi! Sorry for the late reply. To be safe I recommend to always use 'aer' as the unit. Unfortunately `Amount` defaults to use 'aergo' if no unit is passed, so `new Aergo(1)` is interpreted as '1 aergo'. But that shouldn't be an issue for '0'... I'll check and fix it asap!

**Comment by graup** [comment]
I fixed the bug in the contract.asTransaction method and added a new test. Was missing a test covering the combination of contract + local signing. (We're still in the process of phasing out the remote signing, so there might be some inconsistencies :-/)

Published as 0.19.1 on npm. Thanks for the report!

As a general recommendation, use either `'0'`, `'0 aer'`, or `new Amount('0 aer')` for amounts passed to the signer/hasher. It's a bit inconsistent, I know :( Will try to improve API and/or docs in this regard.

**Comment by aspiers** [comment]
Excellent, thanks a lot Paul!! :sunglasses: 

Yes it would be great to improve the docs at least regarding not just these units, but also regarding the combination of contract calls with local signing.  We had to do a bit of guesswork on that front, although having @hanlsin on the team helped a lot of course :slightly_smiling_face:
